testcontainers-common/src/main/java/com/playtika/test/common/operations/DefaultNetworkTestOperations.java

```diff
@@ -0,0 +1,43 @@
+package com.playtika.test.common.operations;
+
+import lombok.RequiredArgsConstructor;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.GenericContainer;
+
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+public class DefaultNetworkTestOperations implements NetworkTestOperations {
+    private final GenericContainer container;
+
+    @Override
+    public void addNetworkLatencyForResponses(long millis) {
+        executeCommandInContainer(() -> {
+                    try {
+                        return container.execInContainer("tc", "qdisc", "add", "dev", "eth0", "root", "netem", "delay", millis + "ms");
+                    } catch (Exception e) {
+                        throw new IllegalStateException("Failed to execute command", e);
+                    }
+                }
+        );
+    }
+
+    @Override
+    public void removeNetworkLatencyForResponses() {
+        executeCommandInContainer(() -> {
+            try {
+                return container.execInContainer("tc", "qdisc", "del", "dev", "eth0", "root");
+            } catch (Exception e) {
+                throw new IllegalStateException("Failed to execute command", e);
+            }
+        });
+    }
+
+    private void executeCommandInContainer(Supplier<ExecResult> command) {
```

Comment:
In order to correspond with the method name you should pass Function<Container, ExecResult> command instead of Supplier<ExecResult>.

Reply:
agreed, makes sense
thanks